lib/puma/plugin.rb

```diff
@@ -62,10 +62,16 @@ def add_background(blk)
     end
 
     def fire_background
-      @background.each do |b|
-        Thread.new do
-          set_thread_name "plugin background"
-          b.call
+      # remove conditional when Ruby 2.2 support is dropped,
+      # no Thread#name method
+      if Thread.current.respond_to?(:name=)
+        @background.each_with_index do |b, i|
```

Comment:
set_thread_name already has code that deals with this, you don't need to do it here.

Reply:
Isn't the fix purely that we should be calling `Puma.set_thread_name`?

Reply:
The only reason `Puma.set_thread_name` exists is to support Ruby 2.2.x.  Come Christmas time...

Reply:
We're going to be supprting 2.2 for the forseeable future so I'd prefer we stick to the method that already exists.

Reply:
Done.